tests/CMakeLists.txt

```diff
@@ -69,16 +137,42 @@ add_custom_command(TARGET libocpp_unit_tests POST_BUILD
 
 add_test(libocpp_unit_tests libocpp_unit_tests)
 
-target_include_directories(libocpp_unit_tests PUBLIC 
-        ${CMAKE_CURRENT_SOURCE_DIR}
-        ${CMAKE_CURRENT_SOURCE_DIR}/lib/ocpp/common
+target_include_directories(libocpp_unit_tests PUBLIC
+        ${TEST_INCLUDE_DIRECTORIES}
 )
 target_link_libraries(libocpp_unit_tests PRIVATE
         ocpp
-        GTest::gmock_main
-        GTest::gtest_main
+        ${GTEST_LIBRARIES}
 )
 
+set(GCOVR_DEPENDENCIES libocpp_unit_tests)
+
+# Add executables, link libraries etc for the unit tests that are separate and do not link against the ocpp lib.
+# This loops over all tests added to `SEPARATE_UNIT_TESTS` and does the necessary things to make it build.
+# For now, everything is added that seems to be needed in every test. If later some sources and link libraries should
+# not be added here, they can always be removed from this loop and added to the test targets that actually need them.
+foreach(ITEM ${SEPARATE_UNIT_TESTS})
+    set(TEST_ROOT_NAME)
+    cmake_path(GET ITEM STEM TEST_ROOT_NAME)
+    set(TEST_NAME "libocpp_${TEST_ROOT_NAME}")
+    add_executable(${TEST_NAME} ${ITEM})
+    add_test(NAME ${TEST_NAME} COMMAND ${TEST_NAME} WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR})
+    target_link_libraries(${TEST_NAME} PUBLIC ${GTEST_LIBRARIES})
+    list(APPEND GCOVR_DEPENDENCIES ${TEST_NAME})
+    target_compile_definitions(${TEST_NAME} PRIVATE ${TEST_COMPILE_DEFINITIONS})
+    target_compile_definitions(${TEST_NAME} PRIVATE
+        MIGRATION_FILE_VERSION_V16=${MIGRATION_FILE_VERSION_V16}
+        MIGRATION_FILE_VERSION_V201=${MIGRATION_FILE_VERSION_V201}
+        MIGRATION_DEVICE_MODEL_FILE_VERSION_V201=${MIGRATION_DEVICE_MODEL_FILE_VERSION_V201})
+    target_compile_options(${TEST_NAME} PRIVATE ${TEST_COMPILE_OPTIONS})
+    target_compile_features(${TEST_NAME} PUBLIC ${TEST_COMPILE_FEATURES})
+    target_include_directories(${TEST_NAME} PRIVATE ${TEST_INCLUDE_DIRECTORIES})
+    target_link_libraries(${TEST_NAME} PRIVATE ${LIBOCPP_TEST_DEFAULT_LINK_LIBRARIES})
+    message("Add test ${TEST_NAME}")
```

Comment:
This could also use `append_coverage_compiler_flags_to_target(${TEST_NAME})` for coverage flags so we can remove the "global" `append_coverage_compiler_flags()` from here: https://github.com/EVerest/libocpp/blob/850bd796e8d4d6984486ef323d1db5cbbddefd80/CMakeLists.txt#L36C5-L36C35